client/tx/query.go

```diff
@@ -48,50 +49,44 @@ func QueryTxCmd(cdc *codec.Codec) *cobra.Command {
 	return cmd
 }
 
-func queryTx(cdc *codec.Codec, cliCtx context.CLIContext, hashHexStr string) ([]byte, error) {
+func queryTx(cdc *codec.Codec, cliCtx context.CLIContext, hashHexStr string) (out sdk.TxResponse, err error) {
 	hash, err := hex.DecodeString(hashHexStr)
 	if err != nil {
-		return nil, err
+		return out, err
 	}
 
 	node, err := cliCtx.GetNode()
 	if err != nil {
-		return nil, err
+		return out, err
 	}
 
 	res, err := node.Tx(hash, !cliCtx.TrustNode)
 	if err != nil {
-		return nil, err
+		return out, err
 	}
 
-	if !cliCtx.TrustNode {
-		err := ValidateTxResult(cliCtx, res)
-		if err != nil {
-			return nil, err
-		}
+	if err = ValidateTxResult(cliCtx, res); !cliCtx.TrustNode && err != nil {
+		return out, err
 	}
 
-	info, err := formatTxResult(cdc, res)
-	if err != nil {
-		return nil, err
+	if out, err = formatTxResult(cdc, res); err != nil {
+		return out, err
 	}
 
-	if cliCtx.Indent {
-		return cdc.MarshalJSONIndent(info, "", "  ")
-	}
-	return cdc.MarshalJSON(info)
+	return out, nil
 }
 
 // ValidateTxResult performs transaction verification
 func ValidateTxResult(cliCtx context.CLIContext, res *ctypes.ResultTx) error {
-	check, err := cliCtx.Verify(res.Height)
-	if err != nil {
-		return err
-	}
-
-	err = res.Proof.Validate(check.Header.DataHash)
-	if err != nil {
-		return err
+	if !cliCtx.TrustNode {
```

Comment:
```go
if cliCtx.TrustNode {
  return nil
}
```

Reply:
this code does this. Slightly confused what you mean here? If we trust the node, we don't want to validate the results.

Reply:
I meant instead of wrapping big chunk of code into if condition, you can return early!